### PR TITLE
Code analysis

### PR DESCRIPTION
**Pull request by agriffard**
Fixes various alerts detected in the code:
https://lgtm.com/projects/g/OrchardCMS/OrchardCore/?mode=list&lang=csharp

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Contents/AdminNodes/ContentTypesAdminNodeNavigationBuilder.cs [hunk 1]
```diff
@@ -90,8 +90,6 @@ private IEnumerable<ContentTypeDefinition> GetContentTypesToShow(ContentTypesAdm
 
             if (!node.ShowAll)
             {
-                node.ContentTypes = node.ContentTypes;
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Contents/AdminNodes/ContentTypesAdminNodeNavigationBuilder.cs:93]: Do we really have such thing ;)

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs [hunk 2]
```diff
@@ -568,9 +568,6 @@ private async Task<IActionResult> ExchangePasswordGrantType(OpenIddictRequest re
 
         private async Task<IActionResult> ExchangeAuthorizationCodeOrRefreshTokenGrantType(OpenIddictRequest request)
         {
-            var application = await _applicationManager.FindByClientIdAsync(request.ClientId) ??
-                throw new InvalidOperationException("The application details cannot be found.");
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs:572]: Not sure abou this change @agriffard 
Why was it being suggested?

Comment (agriffard) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs:572]: Because `application` is not used later int he code.

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs:572]: Remove the variable then, not the check that it exists. (I'm guessing here, but we wouldn't want to lose the behaviour of checking that the client id exists, I think)
i.e.
```
await _applicationManager.FindByClientIdAsync(request.ClientId) ??
                throw new InvalidOperationException("The application details cannot be found.");
```

Comment (agriffard) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs]: re-added but it requires to set a variable.

Comment (kevinchalet) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs]: Feel free to remove it: OpenIddict itself ensures the application exists and this second call will basically return a cached value anyway. If we don't use it, there's no point doing that (we don't allow disabling OpenIddict's L2-like cache anyway).

**Review by deanmarcussen — APPROVED** [review]

**Pull request merged**